File: Backend/BankDBFirstLib/transactionDetails.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BankDBFirstLib
{
    public class transactionDetails
    {
        [Key]
        public int Transaction_id { get; set; }

        public string? TransactionType { get; set; }

        public double? Transaction_amount { get; set; }

        public DateTime? Transaction_time { get; set; }
        public int Customer_id { get; set; }

        public virtual ICollection<transaction_credit> Transaction_credit { get; } = new List<transaction_credit>();

        public virtual ICollection<transaction_debit> TransactionDebits { get; } = new List<transaction_debit>();

    }
}

File: user-portal/src/app/Models/transaction-details.ts
export class TransactionDetails {
    transactionId? : number = 0;
    transactionType: string = ""
    transaction_amount : number =0
    transaction_time : Date = new Date()
    customer_id : number = 0;
    constructor(transactionId: number, transactionType: string, transaction_amount: number, transaction_time : Date, customer_id: number)
    {
        this.transactionId = transactionId;
        this.transactionType = transactionType;
        this.transaction_amount = transaction_amount;
        this.transaction_time = transaction_time;
        this.customer_id = customer_id;
    }
    
}

